frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/selectors.ts

```diff
@@ -0,0 +1,114 @@
+import { iGetIn } from '../../../../../utils/immutable';
+import {
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_WORKLOAD_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_FLAVOR_LABEL,
+} from '../../../../../constants/vm';
+import {
+  CommonTemplatesValidation,
+  IntegerValidtionType,
+  CommonTemplatesValidationAttr,
+  IntegerValidationObj,
+  IntegerBetween,
+} from './validation-types';
+
+export const getRelevantTemplates = (commonTemplates, os, workloadProfile, flavor) => {
+  if (!commonTemplates || commonTemplates.length === 0 || !os || !workloadProfile) {
+    return [];
+  }
+  return (commonTemplates || []).filter(
+    (template) =>
+      iGetIn(template, ['metadata', 'labels', TEMPLATE_TYPE_LABEL]) === 'base' &&
+      (!os || iGetIn(template, ['metadata', 'labels', `${TEMPLATE_OS_LABEL}/${os}`])) &&
+      (!workloadProfile ||
+        iGetIn(template, [
+          'metadata',
+          'labels',
+          `${TEMPLATE_WORKLOAD_LABEL}/${workloadProfile}`,
+        ])) &&
+      (flavor === 'Custom' ||
+        iGetIn(template, ['metadata', 'labels', `${TEMPLATE_FLAVOR_LABEL}/${flavor}`])),
+  );
+};
+
+export const getValidationsFromTemplates = (templates): CommonTemplatesValidation[][] => {
```

Comment:
I think I'm missing something here - why we do not flatten the rules ?

for example, if you have two templates:
template1:
``` json
[
          {
            "name": "core-limits",
            "path": "jsonpath::.spec.domain.cpu.cores",
            "message": "cpu cores must be limited",
            "min": 1,
            "max": 8
          }
        ]

```
template2:
``` json
[
    {
      "name": "validation-rule-01",
      "valid": "jsonpath::.some.json.path",
      "path”: "jsonpath::.some.json.path[*].leaf",
      "rule”: "integer",
      "message”: ".some.json.path[*].leaf must exists",
      "min”: 1,
      "justWarning": true
    },
    {
      "name": "validation-rule-02",
      "valid": "jsonpath::.another.json.path",
      "path": "jsonpath::.another.json.path.item",
      "rule": "integer",
      "message": "jsonpath::./another.json.path.item must be below a threshold",
      "max": "jsonpath::.yet.another.json.path.defines.the.limit"
    }
  ]
```

we should end up with a flat 3 rules and not with 2 sets one having 1 rule and the other 2.

Reply:
Multiple templates can occur when for example the workload profile is not selected right? Only one template will be selected at the end. 
I think we should pick the most relaxed validation (meaning allowing everything if one templates is missing the validation. Or combining min max of the other ones to get the largest interval span).

We should also probably use the validation from the user template if one is selected.

Reply:
First of all, the user must pick the `OS` and the `workload profile` in order to get the relevant templates from common templates. Only the `flavor` can be  selected as `Custom`.

Second, when a user pick a flavor, we don't do any validations to the memory - since it is hard-coded in the template. when the user chooses `Custom` we are getting multiple templates, hence multiple validations from each template. 
The idea here is to pass the validation if at least one of the validations had passed, and fail if none had passed.

`user template` isn't in the scope of this feature and will/should be addressed in another PR.

Reply:
> user template isn't in the scope of this feature and will/should be addressed in another PR.

are you sure ? IFAIU it's the simplest test, where you have one template and do not need to look for any other ...
```
getRelevantTemplates = customTemplate ? [ customTemplate ] | ....
```

Reply:
I've Added support for user template validations